plugins/modules/database/influxdb/influxdb_retention_policy.py

```diff
@@ -148,28 +180,9 @@ def parse_duration_literal(value, extended=False):
     lookup = (EXTENDED_DURATION_REGEX if extended else DURATION_REGEX).findall(value)
 
     for duration_literal in lookup:
-        if extended and duration_literal[3] == 's':
-            duration_val = float(duration_literal[2])
-            duration += duration_val * 1000 * 1000 * 1000
-        else:
-            duration_val = int(duration_literal[0])
-
-            if duration_literal[1] == 'ns':
-                duration += duration_val
-            elif duration_literal[1] == 'u' or duration_literal[1] == 'µ':
-                duration += duration_val * 1000
-            elif duration_literal[1] == 'ms':
-                duration += duration_val * 1000 * 1000
-            elif duration_literal[1] == 's':
-                duration += duration_val * 1000 * 1000 * 1000
-            elif duration_literal[1] == 'm':
-                duration += duration_val * 1000 * 1000 * 1000 * 60
-            elif duration_literal[1] == 'h':
-                duration += duration_val * 1000 * 1000 * 1000 * 60 * 60
-            elif duration_literal[1] == 'd':
-                duration += duration_val * 1000 * 1000 * 1000 * 60 * 60 * 24
-            elif duration_literal[1] == 'w':
-                duration += duration_val * 1000 * 1000 * 1000 * 60 * 60 * 24 * 7
+        filtered_literal = list(filter(None, duration_literal))
+        duration_val = float(filtered_literal[0])
+        duration += duration_val * DURATION_UNIT_NANOSECS[filtered_literal[1]]
```

Comment:
This seems to be quite a different logic than what you had before. Could you please explain the change? I was expcting to see something like:
```suggestion
        if extended and duration_literal[3] == 's':
            duration_val = float(duration_literal[2])
        else:
            duration_val = int(duration_literal[0])
        duration += duration_val * DURATION_UNIT_NANOSECS[filtered_literal[1]]
```

Reply:
Just tried to simplify code to avoid the `if` statement.

Due to how the regex is parsed, either `('1', 'm', '', '')` or `('', '', '1', 's')` tuples are returned. With `filtered_literal = list(filter(None, duration_literal))` empty strings are removed, so final tuple is always `('1', 's')`. This time I have tested this both in Python 2 and Python 3 😅 

If you think this is not clear enough, I can revert to the original `if` as you suggest.

Reply:
Oh OK, I hadn't examined it deeply enough. I think it is good as it is.